### PR TITLE
Filled in Part 1 a

### PR DESCRIPTION
**Pull request by VedantKalbag**

### CODE REVIEW

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 1]
```diff
@@ -20,67 +20,96 @@ You are going to write 10 UDTs in project3.
...
+ */
+
+/*
+Thing 1) CPU
+5 properties:
+    1) number of cores
+    2) clock speed
+    3) L1 cache
+    4) L2 cache
+    5) L3 cache
```
Comment (matkatmusic) [inline main.cpp]: These are objects, not primitives!   You need to pick things that can be represented with `int float double bool char std::string`
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (VedantKalbag) [reply main.cpp]: Hmm.. My thought process was that these could be represented by an integer, eg. int l1Cache = 64. I can still change this if you'd like, but does that make sense?

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 2]
```diff
@@ -108,7 +112,19 @@ Thing 4) Coffee shop
...
 
-
+/*
+Thing 10) Pet Shelter
+5 properties:
+    1) Dog
+    2) Cat
+    3) Bird
+    4) Lizard
+    5) Snake
```
Comment (matkatmusic) [inline main.cpp]: though arrays and inheritance haven't been covered yet, all of these are instances of 'Animal', and will have the same properties, but with different values. 
if arrays had been covered, all of these could live in a single property `std::vector<Animal> animals;` leaving you with 4 more properties to fill in.  
come up with a better object whose sub-objects aren't so similar.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — COMMENTED** [review]
I'll add the next part, but these aren't very good choices for turning into subobjects.  feel free to come up with a new `#10` after you look at the next set of instructions.  no requirement to replace what you have now, though..

**Review by matkatmusic — APPROVED** [review]

**Review by matkatmusic — CHANGES_REQUESTED** [review]
You're missing a nested class.
```
5) make 2 of the 10 user-defined types have a nested class.  
    - this nested class also needs 5 properties and 3 actions.
    - these nested classes are not considered one of your 10 UDTs.
    - this nested class must be related to the class it is nested inside
```
add a nested class. you only have 1.
- [x] revise

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**